ankaios_sdk/_components/manifest.py

```diff
@@ -0,0 +1,166 @@
+
+        Returns:
+            Manifest: An instance of the Manifest class with the loaded data.
+
+        Raises:
+            FileNotFoundError: If the file does not exist.
+            yaml.YAMLError: If there is an error parsing the YAML file.
+        """
+        try:
+            with open(file_path, 'r', encoding="utf-8") as file:
+                return Manifest.from_string(file.read())
+        except Exception as e:
+            raise ValueError(f"Error reading manifest file: {e}") from e
+
+    @staticmethod
+    def from_string(manifest: str) -> 'Manifest':
+        """
+        Creates a Manifest instance from a YAML string.
+
+        Args:
+            manifest (str): The YAML string representing the manifest.
+
+        Returns:
+            Manifest: An instance of the Manifest class with the parsed data.
+
+        Raises:
+            ValueError: If there is an error parsing the YAML string.
+        """
+        try:
+            return Manifest.from_dict(yaml.safe_load(manifest))
+        except Exception as e:
+            raise ValueError(f"Error parsing manifest: {e}") from e
+
+    @staticmethod
+    def from_dict(manifest: dict) -> 'Manifest':
+        """
+        Creates a Manifest instance from a dictionary.
+
+        Args:
+            manifest (dict): The dictionary representing the manifest.
+
+        Returns:
+            Manifest: An instance of the Manifest class with the given data.
+        """
+        return Manifest(manifest)
+
+    def check(self) -> bool:
+        """
+        Validates the manifest data.
+
+        Returns:
+            bool: True if the manifest data is valid, False otherwise.
+        """
+        if "apiVersion" not in self._manifest.keys():
+            return False
+        if "workloads" not in self._manifest.keys():
+            return False
+        wl_allowed_keys = ["runtime", "agent", "restartPolicy",
+                           "runtimeConfig", "dependencies", "tags",
+                           "controlInterfaceAccess"]
+        for wl_name in self._manifest["workloads"]:
+            for key in self._manifest["workloads"][wl_name].keys():
+                if key not in wl_allowed_keys:
+                    return False
+        return True
+
+    def _calculate_masks(self) -> list[str]:
+        """
+        Calculates the masks for the workloads in the manifest.
+
+        Returns:
+            list[str]: A list of masks for the workloads.
+        """
+        return [f"desiredState.workloads.{key}"
+                for key in self._manifest["workloads"].keys()]
+
+    def generate_complete_state(self) -> CompleteState:
+        """
+        Generates a CompleteState instance from the manifest.
+
+        Returns:
+            CompleteState: An instance of the CompleteState class
+                populated with the manifest data.
+        """
+        complete_state = CompleteState()
+        complete_state._from_dict(self._manifest)
+        return complete_state
```

Comment:
So the change would involve:
* deleting the `Manifest.generate_complete_state`
* instead of having the manifest import the complete_state, import the other way around
* rename the `CompleteState._from_dict` to `from_manifest` (or something like this), make it public and make it accept a `Manifest object`.
Is this ok?

Reply:
Done.